cdap-common/src/main/java/co/cask/cdap/common/discovery/RandomEndpointStrategy.java

```diff
@@ -20,12 +20,13 @@
 
 import java.util.Iterator;
 import java.util.Random;
+import java.util.concurrent.ThreadLocalRandom;
 
 /**
  * Randomly picks endpoint from the list of available endpoints.
  */
 public final class RandomEndpointStrategy extends AbstractEndpointStrategy {
-
+  private Random random = ThreadLocalRandom.current();
```

Comment:
This doesn't work. You have to call current from the active thread, I.e. Inside the pick() method